### ISSUE TITLE
fastify websocket handler with typescript

### ISSUE THREAD

**Issue by ExpressGradient** [fastify/fastify-websocket#133]
## 💬  Question here 
How to implement a websocket route using fastify-websocket in typescript?

## Your Environment

- *node version*: 14.15
- *fastify version*: 3.17
- *os*: Windows

**Comment by airhorns** [comment]
> We are missing the module augmentation for RouteShorthandOptionsWithHandler .

Is this type new or something? The module augmentations that add the overloads were working fine as of a few months ago.

**Comment by RafaelGSS** [comment]
> > We are missing the module augmentation for RouteShorthandOptionsWithHandler .
> 
> Is this type new or something? Or has it just been missing the whole time? The module augmentations that add the overloads were working fine as of a few months ago.

The module augmentations for WebSockets is not working AFAIK, look at:

```ts
import type {
  FastifyRequest,
  FastifyReply,
  RouteShorthandOptionsWithHandler,
} from 'fastify';
import { SocketStream } from 'fastify-websocket';


const route1: RouteShorthandOptionsWithHandler = {
  schema: {},
  preValidation: [],
  handler: async (
    _req: FastifyRequest,
    reply: FastifyReply,
  ): Promise<void> => {
    reply.send('works fine');
  },
};

const route2: RouteShorthandOptionsWithHandler = {
  schema: {},
  websocket: true,
  handler: (
    connection: SocketStream,
    _request: FastifyRequest,
  ): void => {
    connection.socket.on('close', () => {});
  },
};
```

We might be missing the augmentation for `RouteShorthandOptionsWithHandler`, we have only the [`RouteShorthandMethod`](https://github.com/fastify/fastify-websocket/blob/master/index.d.ts#L24).

Also, for your purpose @ExpressGradient it should work:

```ts
fastify.get('/example', { websocket: true }, async function(connection: SocketStream, request: FastifyRequest) {
  // ...
});
```

**Comment by ExpressGradient** [comment]
@RafaelGSS Thank You.
I was facing these issues only when I use the `app.route()` with websocket.
There are no problems with `app.get()`.

**Comment by kibertoad** [comment]
@RafaelGSS Any chance you could add the missing augmentation? I gave it a shot, but couldn't find any example of this kind of augmentation in any existing plugin, and our TS type system is very, very complex :)

**Comment by RafaelGSS** [comment]
> @RafaelGSS Any chance you could add the missing augmentation? I gave it a shot, but couldn't find any example of this kind of augmentation in any existing plugin, and our TS type system is very, very complex :)

Sure, I'll do it tomorrow!

**Comment by RafaelGSS** [comment]
I gave it a shot as well but looks like we should change our internal types to satisfy handler augmentation on `RouteShorthandOptionsWithHandler`. For folks who want to give a try, this is the test case:

```ts
app.get('/example', {
  websocket: true,
  handler: (connection, request) => {
    expectType<SocketStream>(connection);
    expectType<FastifyRequest<RequestGenericInterface>>(request);
  },
})
```

cc: @fastify/typescript

**Comment by airhorns** [comment]
Fixed in #128!
